sale_import_base/tests/test_sale_order_import.py

```diff
@@ -0,0 +1,158 @@
+#  Copyright (c) Akretion 2020
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+import json
+
+from odoo.exceptions import ValidationError
+
+from .common_sale_order_import import SaleImportCase
+
+
+class TestSaleOrderImport(SaleImportCase):
+    def setUp(self):
+        super().setUp()
+        self.sale_channel_ebay = self.env.ref("sale_channel.sale_channel_ebay")
+
+    def test_invalid_json(self):
+        """ An invalid input will stop the job """
+        data = self.sale_data
+        del data["address_customer"]["street"]
+        with self.assertRaises(ValidationError):
+            self.importer_component.run(json.dumps(data))
+
+    def test_create_partner(self):
+        """ Base scenario: create partner """
+        data = self.sale_data
+        partner_count = (
+            self.env["res.partner"].with_context(active_test=False).search_count([])
+        )
+        self.importer_component.run(json.dumps(data))
+        partner_count_after_import = (
+            self.env["res.partner"].with_context(active_test=False).search_count([])
+        )
+        self.assertEqual(partner_count_after_import, partner_count + 3)
+
+    def test_binding_created(self):
+        """ When we create a partner, a binding is created """
+        data = self.sale_data
+        new_sale_order = self.importer_component.run(json.dumps(data))
+        binding_count = self.env["sale.channel.partner"].search_count(
+            [
+                ("sale_channel_id", "=", self.sale_channel_ebay.id),
+                ("partner_id", "=", new_sale_order.partner_id.id),
+                ("external_id", "=", "ThomasJeanEbay"),
+            ]
+        )
+        self.assertEqual(binding_count, 1)
+
+    def test_import_existing_partner_match_external_id(self):
+        """ During import, if a partner is matched on external_id/channel
+        combination, his address is updated """
+        data = self.sale_data
+        partner_colleen = self.env.ref("base.res_partner_address_28")
+        del data["address_customer"]["email"]
```

Comment:
instead of changing the json of data. I would do something more explicite
partner = self.env.ref("...")
self.env["sale.channel.partner"].create({"partner_id": partner.id, "external_id": "ThomasJeanEbay", "sale_channel_id", "=", self.sale_channel_ebay.id})
self.importer_component.run(json.dumps(data))
self.assertEqual(partner.street, "....")